boot/freeldr/freeldr/arch/uefi/arch/i386/uefiasm.S

```diff
@@ -0,0 +1,84 @@
+/*
+ * PROJECT:     FreeLoader UEFI Support
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     UEFI assembly exit code
+ * COPYRIGHT:   Copyright 2023 Justin Miller <justinmiller100@gmail.com>
+ */
+
+#include <asm.inc>
+#include <arch/pc/x86common.h>
+#include <arch/pc/pcbios.h>
+
+.code32
+PUBLIC _gdtptr
+PUBLIC _i386idtptr
+EXTERN _i386Idt:DWORD
+EXTERN _EndofExitStack:DWORD
```

Comment:
I was looking at existing code in master. This stack is a new piece of memory allocated in UefiPrepareForReactOS() with LoaderOsloaderStack type. I was wondering: When exiting the UEFI boot services, what happens to the existing stack FreeLdr is using? Is it still valid? If so, why wouldn't it be possible to use that latter one instead of a new one?

Reply:
Thinking more about that, in PC-BIOS architecture we reserve when freeldr starts, a zone for the freeldr stack:
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/arch/i386/pc/pcmem.c;hb=f18111b64120806529e93404f21b6320f211ec39#l546
It may be worth investigating that and doing something similar for UEFI too. The stack used by FreeLdr there should be already known in the main() function.

Reply:
Problem is you still have to have the UEFI provided stack In order to exit boot services. So I can’t just change it before calling the main function

Reply:
It looks like now you can use a way similar to the one in AMD64, using a parameter of the `__ExitUefi` function to pass the end of exit stack, and getting rid of this `_EndofExitStack` variable.